### PR TITLE
Improve metric collection and metrics collected 

### LINKED ISSUE #162: Improve debug metrics
SPS and UPS are opaque (even forgot what they stand for) and not very accurate because they averages over all of time making it hard to notice change. They should probably average over the last minute or something. 
We should also have metrics to track server and networking performance.

### PR DESCRIPTION
**Pull request by SourenP**
**Description**

- New TemporalMetric class that helps accurately track metrics over a span of time
- Track metrics FPS, UPS, XPS (network updates per second), TDA (tick delta average), TIK count
- Include Boost

Screen Shot 2020-01-09 at 5 25 06 PM

**Fixes**

Fixes #162

### CODE REVIEW

File: src/client/systems/ClientGameController.hpp [hunk 1]
```diff
@@ -69,7 +69,8 @@ class ClientGameController : public GameController {
     uint32_t m_playerId = 0;
     bool m_headless, m_isBot;
     BotStrategy m_strategy;
-    std::string m_server_ip;
+    std::string m_serverIP;
+    uint m_applyServerStateCount = 0;
```
Comment (SourenP) [inline src/client/systems/ClientGameController.hpp]: delete this

File: src/common/systems/GameController.hpp [hunk 2]
```diff
@@ -50,8 +53,9 @@ class GameController {
     sf::Clock m_clock;
     sf::Int32 m_elapsedTime = 0;
     sf::Int32 m_timeAccumulator = 0;
-    sf::Int32 m_stepCount = 0;
-    sf::Int32 m_updateCount = 0;
+
+    TemporalMetric m_stepMetric{120, sf::seconds(0.5f)};
+    TemporalMetric m_updateMetric{120, sf::seconds(0.5f)};
```
Comment (SourenP) [inline src/common/systems/GameController.hpp]: rename these to be more specific
maybe `m_gameStepMetric` and `m_gameUpdateMetric `

File: src/client/systems/ClientGameController.cpp [hunk 3]
```diff
@@ -71,14 +69,12 @@ void ClientGameController::preUpdate() {
         }
         case GameStatus::playing: {
             if (m_isBot) {
+                // Override user input with bot input
                 inputs = m_gameObjectController->getBotMove(m_playerId, m_strategy);
             }
             sendInputs(inputs);
             saveInputs(inputs);
-
-            if (m_networkingClient->getGameStateIsFresh()) {
```
Comment (copacetic) [inline src/client/systems/ClientGameController.cpp]: why delete this?

File: src/common/metrics/TemporalMetric.hpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * This class currenly only supports uint counts.
+ */
+
+#include <SFML/System.hpp>
+#include <boost/circular_buffer.hpp>
+
+#ifndef TemporalMetric_hpp
+#define TemporalMetric_hpp
+
+class TemporalMetric {
```
Comment (copacetic) [inline src/common/metrics/TemporalMetric.hpp]: can you add some tests for this? It would suck if we got it wrong

Comment (SourenP) [reply src/common/metrics/TemporalMetric.hpp]: I was going to but then I realized how difficult it would be without mocking time.
Any ideas?

**Pull request merged**